packages/client/src/base/tool-manager/tool-manager.ts

```diff
@@ -75,27 +79,23 @@ export interface IToolManager {
  * registration of tools via Dependency Injection.
  */
 @injectable()
-export class ToolManager implements IToolManager {
-    readonly actives: Tool[] = [];
+export class ToolManager implements IToolManager, IDiagramStartup {
+    @inject(EditorContextService)
+    protected editorContext: EditorContextService;
+
+    @inject(LazyInjector)
+    protected readonly lazyInjector: LazyInjector;
 
-    protected editorContext?: EditorContextService;
-
-    constructor(
-        @multiInject(TYPES.ITool)
-        @optional()
-        readonly tools: Tool[] = [],
-        @multiInject(TYPES.IDefaultTool)
-        @optional()
-        readonly defaultTools: Tool[],
-        @inject(TYPES.IEditorContextServiceProvider) protected contextServiceProvider: EditorContextServiceProvider
-    ) {
-        this.registerTools(...this.tools);
-        this.registerDefaultTools(...this.defaultTools);
+    readonly actives: Tool[] = [];
+    readonly tools: Tool[] = [];
+    readonly defaultTools: Tool[] = [];
+
+    preLoadDiagram(): MaybePromise<void> {
+        const tools: Tool[] = this.lazyInjector.getAll(TYPES.ITool);
+        const defaultTools: Tool[] = this.lazyInjector.getAll(TYPES.IDefaultTool);
+        this.registerTools(...tools);
+        this.registerDefaultTools(...defaultTools);
         this.enableDefaultTools();
-        this.contextServiceProvider().then(editorContext => {
-            editorContext.onEditModeChanged(change => this.editModeChanged(change.newValue, change.oldValue));
```

Comment:
Was this removed by accident? I don't see how our `editModeChanged` method will be triggered now and the ToolManager is not bound as a `IEditModeListener`.

Reply:
Good catch. Yes, the tool manager should be bound as an IEditModeListener Instead

Reply:
Tested and works again, thank you!